server/cmwell-data-tools/src/main/scala/cmwell/tools/data/downloader/DataPostProcessor.scala

```diff
@@ -45,11 +45,14 @@ object DataPostProcessor extends DataToolsLogging {
           false
         case _ => true
       }
-      .groupBy(maxSubjects + 1, GroupChunker.extractSubject) // + 1: see implementation of groupBy ...
-      .fold(Seq.empty[ByteString]){(agg, b) => agg :+ b} // todo: check if endl is needed here
-      .map(concatByteStrings(_, endl))
-      .map(_ ++ endl)
-      .mergeSubstreams
+      .fold(Map.empty[ByteString, ByteStringBuilder]){ (agg, line) =>
+        // aggregate each line according to its subject (i.e., bucket)
+        val subject = GroupChunker.extractSubject(line)
+        val builder = agg.getOrElse(subject, new ByteStringBuilder)
+        builder ++= (line ++ endl)
+        agg + (subject -> builder)
```

Comment:
using immutable map that creates a inner array for *each* line is too time consuming.